frescobaldi_app/panelmanager.py

```diff
@@ -45,33 +47,49 @@ def __init__(self, mainwindow):
 
         """
         self._panels = []
+        self._submenus = collections.OrderedDict([
+            ('viewers', {
+                'menu': QMenu(_('&Viewers')),
+                'panels': []
+            }),
+            ('coding', {
+                'menu': QMenu(_('&Coding')),
+                'panels': []
+            }),
+            ('structure', {
+                'menu': QMenu(_('&Structure')),
+                'panels': []
+            }),
+            ])
 
         # add the the panel stubs here
-        self.loadPanel("quickinsert.QuickInsertPanel")
-        self.loadPanel("musicview.MusicViewPanel")
-        self.loadPanel("svgview.SvgViewPanel")
-        self.loadPanel("viewers.manuscript.ManuscriptViewPanel")
-        self.loadPanel("logtool.LogTool")
-        self.loadPanel("docbrowser.HelpBrowser")
-        self.loadPanel("snippet.tool.SnippetTool")
+        self.loadPanel("musicview.MusicViewPanel", "viewers")
+        self.loadPanel("svgview.SvgViewPanel", "viewers")
+        self.loadPanel("viewers.manuscript.ManuscriptViewPanel", "viewers")
+        self.loadPanel("docbrowser.HelpBrowser", "viewers")
+        self.loadPanel("logtool.LogTool", "viewers")
+        self.loadPanel("quickinsert.QuickInsertPanel", "coding")
+        self.loadPanel("charmap.CharMap", "coding")
+        self.loadPanel("snippet.tool.SnippetTool", "coding")
+        self.loadPanel("doclist.DocumentList", "structure")
+        self.loadPanel("outline.OutlinePanel", "structure")
         self.loadPanel("miditool.MidiTool")
         self.loadPanel("midiinput.tool.MidiInputTool")
-        self.loadPanel("charmap.CharMap")
-        self.loadPanel("doclist.DocumentList")
-        self.loadPanel("outline.OutlinePanel")
         self.loadPanel("layoutcontrol.LayoutControlOptions")
 
         # The Object editor is highly experimental and should be
         # commented out for stable releases.
         if app.is_git_controlled() or QSettings().value("experimental-features", False, bool):
-            self.loadPanel("objecteditor.ObjectEditor")
+            self.loadPanel("objecteditor.ObjectEditor", "coding")
         self.createActions()
 
         # make some default arrangements
         mainwindow.tabifyDockWidget(self.musicview, self.docbrowser)
         mainwindow.tabifyDockWidget(self.musicview, self.svgview)
 
-    def loadPanel(self, name):
+        print(self._submenus)
```

Comment:
@uliska Please, remember to delete debugging lines like this one before committing.

Reply:
Arggh